src/pubsub/ua_pubsub_networkmessage.c

```diff
@@ -1385,6 +1386,14 @@ UA_DataSetMessage_encodeBinary(const UA_DataSetMessage* src, UA_Byte **bufPos,
         return UA_STATUSCODE_BADNOTIMPLEMENTED;
     }
 
+    /* if the configuredSize is set -> padd the message with 0. */
+    if(src->configuredSize > 0 && src->header.dataSetMessageValid) {
+        size_t padding = (size_t)(bufEnd - *bufPos);
+        /* Set the bytes to 0*/
+        memset(*bufPos, 0, padding);
+        /* move the bufpos accordingly*/
+        *bufPos += padding;
```

Comment:
*bufPos = bufEnd

Reply:
I used `UA_Byte_copy(bufEnd, *bufPos);` since bufEnd is const and `*bufPos = bufEnd` results in `error: assignment discards ‘const’ qualifier `